src/nav/AppNavigator.js

```diff
@@ -119,7 +120,19 @@ export default function AppNavigator(props: Props) {
       <Stack.Screen name="account-details" component={AccountDetailsScreen} />
       <Stack.Screen name="group-details" component={GroupDetailsScreen} />
       <Stack.Screen name="auth" component={AuthScreen} />
-      <Stack.Screen name="chat" component={ChatScreen} />
+      <Stack.Screen
+        name="chat"
+        component={ChatScreen}
+        options={{
+          header: stackHeaderProps => {
+            const { route } = stackHeaderProps.scene;
+            // $FlowFixMe - Flow doesn't know what route we're on
+            const params: RouteParamsOf<typeof ChatScreen> = route.params;
+            const { editMessage, narrow } = params;
+            return <ChatNavBar editMessage={editMessage} narrow={narrow} />;
```

Comment:
Hmm -- it seems like a bit of a step backward for the details of this definition to appear here at the navigator, rather than next to the `ChatScreen` component. In particular it makes it rather indirect to be looking at `ChatScreen` and work out where the header is coming from.

I think one pretty reasonable solution would be to put this code over in `ChatScreen.js`, either as a separately exported function or perhaps better as a property on `ChatScreen`. Like `ChatScreen.header = (params: RouteParamsOf<typeof ChatScreen>) => { … };`.

Even then, though, that still seems more complicated than just invoking `ChatNavBar` at the top of the component's own rendered tree. Does this buy us something that's worth the complexity?

(further discussion in main thread)